### PR TITLE
Initial commit for upgrade.

### PR DESCRIPTION
**Pull request by philip-swrve**
Upgrade to 3.0 - take 2.

### CODE REVIEW

File: packages/react-scripts/config/webpack.config.js [hunk 1]
```diff
@@ -344,10 +406,13 @@ module.exports = function(webpackEnv) {
               test: [/\.bmp$/, /\.gif$/, /\.jpe?g$/, /\.png$/],
               loader: require.resolve('url-loader'),
               options: {
-                limit: 10000,
+                limit: imageInlineSizeLimit,
+                // Swrve (START): Support media file location resolution.
                 publicPath: isEnvProduction ? '../media' : '/static/media',
                 outputPath: 'static/media',
                 name: '[name].[hash:8].[ext]',
+                // name: 'static/media/[name].[hash:8].[ext]',
```
Comment (stevenfitzpatrick) [inline packages/react-scripts/config/webpack.config.js:414]: This name comment needed ?

**Comment by stevenfitzpatrick** [comment]
Looks good just left some minor comments

File: packages/react-scripts/config/webpack.config.js (hunk 1 above)
Comment (philip-swrve) [reply packages/react-scripts/config/webpack.config.js:414]: Yes, just kept it to allow reader know what was originally there. All other changes in this file are additions or commented out code, however, this specific line has actually been modified from its original form.

**Comment by stevenfitzpatrick** [comment]
🚀

**Pull request merged**